engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformula/ObjectRollingFormulaOperator.java

```diff
@@ -0,0 +1,195 @@
+                    Collections.singletonMap(PARAM_COLUMN_NAME, formulaInputSource));
+            formulaOutputSource = formulaCopy.getDataView();
+
+            windowValues = new ObjectRingBuffer<>(BUFFER_INITIAL_CAPACITY, true);
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            windowValues = null;
+        }
+
+        @Override
+        public void setValueChunks(@NotNull final Chunk<? extends Values>[] valueChunks) {
+            influencerValuesChunk = valueChunks[0].asObjectChunk();
+        }
+
+        @Override
+        public void accumulateRolling(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] influencerValueChunkArr,
+                @Nullable final LongChunk<OrderedRowKeys> affectedPosChunk,
+                @Nullable final LongChunk<OrderedRowKeys> influencerPosChunk,
+                @NotNull final IntChunk<? extends Values> pushChunk,
+                @NotNull final IntChunk<? extends Values> popChunk,
+                final int len) {
+
+            setValueChunks(influencerValueChunkArr);
+            setPosChunks(affectedPosChunk, influencerPosChunk);
+
+            int pushIndex = 0;
+
+            // chunk processing
+            for (int ii = 0; ii < len; ii++) {
+                final int pushCount = pushChunk.get(ii);
+                final int popCount = popChunk.get(ii);
+
+                if (pushCount == NULL_INT) {
+                    writeNullToOutputChunk(ii);
+                    continue;
+                }
+
+                // pop for this row
+                if (popCount > 0) {
+                    pop(popCount);
+                }
+
+                // push for this row
+                if (pushCount > 0) {
+                    push(pushIndex, pushCount);
+                    pushIndex += pushCount;
+                }
+
+                // write the results to the output chunk
+                writeToOutputChunk(ii);
+            }
+
+            // chunk output to column
+            writeToOutputColumn(inputKeys);
+        }
+
+        @Override
+        public void push(int pos, int count) {
+            windowValues.ensureRemaining(count);
+
+            for (int ii = 0; ii < count; ii++) {
+                final T val = influencerValuesChunk.get(pos + ii);
+                windowValues.addUnsafe(val);
+            }
+        }
+
+        @Override
+        public void pop(int count) {
+            Assert.geq(windowValues.size(), "charWindowValues.size()", count);
+
+            for (int ii = 0; ii < count; ii++) {
+                windowValues.removeUnsafe();
+            }
+        }
+
+        @Override
+        public void writeToOutputChunk(int outIdx) {
+            // If not empty (even if completely full of null), run the formula over the window values.
+            formulaInputSource.set(new ObjectVectorDirect<>(windowValues.getAll()));
```

Comment:
Any reason to not introduce an `ObjectRingBufferVectorWrapper`? It would save on some garbage generation.

Reply:
I thought I had a reason, but pretty sure I'm wrong. Good catch.